content/en/real_user_monitoring/guide/upload-javascript-source-maps.md

```diff
@@ -0,0 +1,103 @@
+
+module.exports = {
+  mode: 'production',
+  devtool: false,
+  plugins: [
+    new webpack.SourceMapDevToolPlugin({
+      noSources: false,
+      filename: '[name].[hash].js.map'
+    }),
+    // ...
+  ],
+  optimization: {
+    minimize: true,
+    // ...
+  },
+  // ...
+};
+```
+
+**Note**: If you are using TypeScript, make sure to set `compilerOptions.sourceMap` to `true` when configuring your `tsconfig.json` file.
+
+{{% /tab %}}
+{{% tab "ParcelJS" %}}
+
+Parcel generates source maps by default when you run the build command: `parcel build <entry file>`
+
+{{% /tab %}}
+{{< /tabs >}}
+
+After building your application, bundlers generate a directory, most of the time named `dist`, with minified Javascript files colocated with their corresponding source maps. Check an example below:
+
+```bash
+./dist
+    javascript.364758.min.js
+    javascript.364758.js.map
+    ./subdirectory
+        javascript.464388.min.js
+        javascript.464388.js.map
+```
+
+## Upload your source maps
+
+The best way to upload source maps is to add an extra-step in your CI pipeline and to run the dedicated command from the [Datadog CLI][2]: it scans the `dist` directory and its subdirectories to automatically upload the source maps with their related minified files. The flow is the following:
+
+1. Add `@datadog/datadog-ci` to your `package.json` file (make sure to use the latest version).
+2. [Create a new and dedicated Datadog API key][3] and export it as an environment variable named `DATADOG_API_KEY`.
+3. Run the following command:
+```bash
+datadog-ci sourcemaps upload /path/to/dist \
+	--service=my-service \
+	--release-version=v35.2395005 \
+	--minified-path-prefix=https://hostname.com/static/js
+```
+
+**Note**: the CLI has been optimised to upload as many source maps as you need in a very short amount of time (typically a few seconds) to minimize the overhead on your CI's performance.
+
+By running this command against our example `dist` directory (see previous section), Datadog will expect your server or your CDN to deliver the javascript files at `https://hostname.com/static/js/javascript.364758.min.js` and `https://hostname.com/static/js/subdirectory/javascript.464388.min`.js.  When an error occurs in a session of one of your users, the RUM SDK instantaneously collects it: whenever the given error originated in a file that were downloaded from one of those urls and is also tagged with `version:v35.2395005` and `service:my-service`, the related source map will be used to deobfuscate the stack trace (in this case, the `javascript.464388.js.map` file).
+
+## Troubleshoot errors with ease
+
+A minified stack trace is almost useless as you don't have access to the file path and the line number: you basically do not know where it is happening in your code base. In addition, the code snippet is still minified (one long line of transformed code) which makes the troubleshooting process even hard. See below an example of an unminified stack trace:
```

Comment:
```suggestion
A minified stack trace is not helpful as you don't have access to the file path and the line number. It's hard to know where something is happening in your code base. In addition, the code snippet is still minified (one long line of transformed code) which makes the troubleshooting process even harder. See below an example of an unminified stack trace:
```

Reply:
I feel this section would be better in the overview. It gives an example of how/why the setup will make their life easier versus actual troubleshooting tips.